src/data/polynomial/eval.lean

```diff
@@ -328,6 +328,10 @@ by rw [← C_1, C_comp]
 @[simp] lemma mul_comp {R : Type*} [comm_semiring R] (p q r : polynomial R) :
   (p * q).comp r = p.comp r * q.comp r := eval₂_mul _ _
 
+lemma prod_comp {R : Type*} [comm_semiring R] (s : multiset (polynomial R)) (p : polynomial R) :
+  s.prod.comp p = (s.map (λ q : polynomial R, q.comp p)).prod :=
+(s.prod_hom (monoid_hom.mk (λ q : polynomial R, q.comp p) one_comp (λ q r, mul_comp q r p))).symm
```

Comment:
I think this `monoid_hom.mk` can be used to prove the lemma below too.

What happens if you redefine `comp` to use `eval₂_ring_hom` instead of `eval₂`? Then the monoid_hom structure will already be there.

Reply:
I've shortened the proof of the pow lemma. Redefining `comp` in terms of `eval₂_ring_hom` would require adding a commutativity assumption.